.github/scripts/format_dbt_test_failures.py

```diff
@@ -189,6 +295,93 @@ def main() -> None:
         date_today = datetime.datetime.today().strftime("%Y-%m-%d")
         output_filepath = f"qc_test_failures_{date_today}.xlsx"
 
+    failed_test_cache_path = get_failed_test_cache_path(
+        run_results_filepath,
+        manifest_filepath,
+    )
+
+    if os.path.isfile(failed_test_cache_path):
+        print(f"Loading failed tests from cache at {failed_test_cache_path}")
+        failed_tests_by_category = get_failed_tests_by_category_from_file(
+            failed_test_cache_path
+        )
+    else:
+        print(
+            f"Failed test cache not found at {failed_test_cache_path}, "
+            "loading failed tests from Athena"
+        )
+        failed_tests_by_category = get_failed_tests_by_category_from_athena(
+            run_results_filepath,
+            manifest_filepath,
+        )
+        print(f"Saving failed tests to the cache at {failed_test_cache_path}")
+        save_failed_tests_by_category_to_file(
+            failed_tests_by_category,
+            failed_test_cache_path,
+        )
+
+    print("Generating the output workbook")
+    # It's important to use a write-only workbook here because otherwise
+    # the metadata required to store cell info about a large number of failing
+    # tests can cause the process to run out of memory
+    workbook = openpyxl.Workbook(write_only=True)
```

Comment:
**question:** Is there a way to "convert" the workbook to read-write on final output? Solely to get rid of the dialog box when initially opening the output.

Reply:
Hm, what dialog box are you seeing? I don't see it when I open the file locally, but perhaps something changes during transfer or upload to OneDrive.

Reply:
Hunh, I thought I remembered a dialog box but don't see anything now after downloading the latest version sent via email. All good!